app/controllers/courses_controller.rb

```diff
@@ -0,0 +1,92 @@
+class CoursesController < ApplicationController
+  before_action :set_course, only: [:show, :edit, :update, :destroy]
+  before_action :authenticate_user!
+  before_action :check_permitted_or_not , only: [:edit , :destroy]
+
+  def index
+    @courses = Course.recent_courses.page params[:page]
+  end
+
+  def show; end
+
+  def new
+    @course = Course.new
+  end
+
+  def edit; end
+
+  def create
+    @course = Course.new　course_params
+    @course.user_id = current_user.id
+
+    respond_to do |format|
+      if @course.save
+        format.html { redirect_to @course, notice: t("controller.course.create") }
+        format.json { render :show, status: :created, location: @course }
+      else
+        format.html { render :new }
+        format.json { render json: @course.errors, status: :unprocessable_entity }
+      end
+    end
+  end 
+
+  def update
+    respond_to do |format|
+      if @course.update(course_params)
+        format.html { redirect_to @course, notice: t("controller.course.updated") }
+        format.json { render :show, status: :ok, location: @course }
+      else
+        format.html { render :edit }
+        format.json { render json: @course.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    @course.destroy
+    respond_to do |format|
+      if @course.destroy
+        format.html { redirect_to courses_path, notice: t("controller.course.destroy") }
+        format.json { head :no_content }
+      else
+        format.html {redirect_to courses_path, notice: t("controller.course.not_destroy")}
+        format.json { head :no_content }
+    end
+  end
+
+  private
+
+  def course_params
+    params.require(:course).permit(:title,:user_id)
+  end
+
+  def set_course
+    @course = Course.find_by(params[:user_id])
```

Comment:
plz, handling when course is nil
add code 
`return if @course`
`flash[:warning] = "Course not found"`
`redirect_to root_path`